### ISSUE TITLE
Replicate the windows shared memory interface

### ISSUE THREAD

**Issue by pushfoo** [openc2e/openc2e#148]
To have a 1:1 drop-in compatibility with Windows c2e, it needs to implement the windows shared memory interface that the CAOS Tool and other applications use.

**Comment by pushfoo** [comment]
I might give this a shot in the coming week or two btw

**Comment by Kirtai** [comment]
It would be nice if such a thing existed on other platforms, maybe a unix socket or something too?

**Comment by pushfoo** [comment]
@Kirtai Yes, absolutely, but it's a separate ticket. Most users are on Windows so it's a lower priority.
I'm also waiting on someone to publish reliable lc2e setup instructions for a current distro so the behavior of the linux version can be better verified. The last time I tried to get lc2e working, it needed some ugly things to be done. Even if I could find those notes, I'm not sure that they're still relevant.

**Comment by Kirtai** [comment]
@pushfoo You weren't kidding about it being ugly, but I got it working and wrote it up [here](https://gist.github.com/Kirtai/6ded27834a822ba835daabcd66f8f4dc)

**Comment by diagonalfish** [comment]
I... kinda dispute the need for this, unless there's a strong demand for using very old tools that require it. Seems like something that you wouldn't want to enable by default, at the very least. Especially since it's not something that would work with multiple instances, or with c2e and openc2e running at the same time.

Openc2e already has a perfectly functional TCP socket interface which works like the one that the DS Linux engine has, and is cross platform.

**Comment by pushfoo** [comment]
> Seems like something that you wouldn't want to enable by default, at the very least

I agree with that. 
However, adding support for it wasn't just about drop-in feature completionism. There was prior discussion in discord around having openc2e also act as a library and ground source of truth that other libraries in interpreted languages could bind. I was hoping that implementations in python and JS could bind the functionality supporting the shared memory and socket interfaces once they were added to openc2e.
